### ISSUE TITLE
bug: Result set Error, when executing left join Clause

### ISSUE THREAD

**Issue by davidshiz** [stoneatom/stonedb#1117]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/stoneatom/stonedb/blob/stonedb-5.7-dev/CONTRIBUTING.md).

### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the problem

```shell
mysql> select count(*) as count2 from customer left join orders on orders.u_custkey = customer.u_custkey;
+--------+
| count2 |
+--------+
|   1547 |
+--------+
1 row in set (0.00 sec)
```

### Expected behavior

```shell
mysql> select count(*) as count2 from customer left join orders on orders.u_cust
key = customer.u_custkey;
+--------+
| count2 |
+--------+
|   1550 |
+--------+
1 row in set (0.01 sec)
```

### How To Reproduce

```sql
create table customer (
        c_custkey int,
        c_name varchar (25),
        c_address varchar (40),
        c_nationkey int,
        c_phone char (15),
        c_acctbal decimal(12,2),
        c_mktsegment char (10),
        c_comment varchar (117)
);

create table orders (
        o_orderkey int,
        o_custkey int,
        o_orderstatus char (1),
        o_totalprice decimal(12,2),
        o_orderdate date,
        o_orderpriority char (15),
        o_clerk char (15),
        o_shippriority int,
        o_comment varchar (79)
);

LOAD DATA LOCAL infile '../1m_customer.tbl' INTO TABLE customer FIELDS TERMINATED BY '|';
LOAD DATA LOCAL infile '../1m_orders.tbl' INTO TABLE orders FIELDS TERMINATED BY '|';

alter table customer add column u_custkey int unsigned;
update customer set u_custkey=c_custkey;
alter table orders add column u_custkey int unsigned;
update orders set u_custkey=o_custkey;

select count(*) as count2 from customer left join orders on orders.u_custkey = customer.u_custkey;
```
The test data is compressed as follows：
[test_data.zip](https://github.com/stoneatom/stonedb/files/10218320/test_data.zip)

### Environment

```shell
root@ub01:/stonedb57/install# /stonedb57/install/bin/mysqld --version
/stonedb57/install/bin/mysqld  Ver 5.7.36-StoneDB for Linux on x86_64 (build-)
build information as follow:
        Repository address: https://github.com/stoneatom/stonedb.git:stonedb-5.7-dev
        Branch name: stonedb-5.7-dev
        Last commit ID: e395f89e1
        Last commit time: Date:   Tue Dec 6 01:04:24 2022 +0800
        Build time: Date: Tue Dec  6 16:03:58 CST 2022
root@ub01:~# cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.5 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.5 LTS"
VERSION_ID="20.04"
```

### Are you interested in submitting a PR to solve the problem?

- [ ] Yes, I will!

**Comment by davidshiz** [comment]
With the same table structure and data, 

tianmu engine:
```shell
mysql> select count(*) from customer left join orders on orders.u_custkey = customer.u_custkey where c_name='Customer#000000001';
+----------+
| count(*) |
+----------+
|        9 |
+----------+
1 row in set (0.01 sec)
```
innodb engine:
```shell
mysql> select count(*) from customer left join orders on orders.u_custkey = customer.u_custkey where c_name='Customer#000000001';
+----------+
| count(*) |
+----------+
|        5 |
+----------+
1 row in set (0.00 sec)
```

**Comment by davidshiz** [comment]
issue closed, the root cause ref #1253
